src/lib.rs

```diff
@@ -285,6 +288,28 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a function for sorting directory entries.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order. The compare function will be called to compare names from entries
+    /// from the same directory. Just the file_name() part of the paths is
+    /// passed to the compare function.
+    /// If no function is set, the entries will not be sorted.
+    ///
+    /// ```rust,no-run
+    /// use std::cmp;
+    /// use std::ffi::OsString;
+    /// use walkdir::WalkDir;
+    ///
+    /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
+    ///     a.cmp(b)
+    /// }
+    /// WalkDir::new("foo").sort_by(string_compare);
```

Comment:
Could you use a closure here instead of a `fn`? I think you needed a `fn` before, but you shouldn't now.